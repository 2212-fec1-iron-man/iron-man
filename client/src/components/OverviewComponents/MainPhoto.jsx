import React from "react";
import {useEffect, useState} from "react";
import axios from "axios";
import Thumbnail from "./Thumbnail.jsx"


const MainPhoto = ({productID, photoNumber, setPhotoNumber, styleNumber}) => {


  const [photoList, setPhotoList] = useState([]);

  useEffect(() => {

    axios.get(`http://localhost:3000/products/${productID}/styles`)
    .then((response) => {
      setPhotoList(response.data.results[styleNumber].photos);
    })
    .catch((error) => {
      console.log('this is an axios get error in MainPhoto.jsx: ', error);
    })
  },[]);


  //TODO: figure out why fifth photo only shows during handleprev.
  const handleNext = function() {
    if (photoNumber === photoList.length - 1) {
      setPhotoNumber(0);
    } else {
      setPhotoNumber(photoNumber++);
    }
  }



  const handlePrev = function() {
    if (photoNumber === 0) {
      setPhotoNumber(photoList.length - 1);
    } else {
      setPhotoNumber(photoNumber--);
    }
  }

  if (photoList.length === 0) {
    return (
      <span>Data not available</span>
    )
  };

  //CSS:
  const arrowStylePrev = {
    position: "absolute",
    top: "50%",
    left: "115px",
    cursor: "pointer",
    width: "auto",
    padding: "0px",
    color: "black",
    fontWeight: "bold",
    fontSize: "38px",
    transition: "0.6s ease",
    borderRadius: "0 3px 3px 0",
    userSelect: "none",
  }

  const arrowStyleNext = {
    position: "absolute",
    top: "50%",
    right: "15px",
    cursor: "pointer",
    width: "auto",
    padding: "0px",
    color: "black",
    fontWeight: "bold",
    fontSize: "38px",
    transition: "0.6s ease",
    borderRadius: "0 3px 3px 0",
    userSelect: "none",
  }

  const imageContainer = {
    backgroundColor: "LightGray",
    position: "relative",
    display: "inline-block",
    width: "800px",
    height: "600px"
  }

  const mainPhoto = {
      width: "100%",
      height: "100%",
  }

  return (
  <>
    <div className="imageContainer" style={imageContainer}>
     <img className="mainPhoto" src={photoList[photoNumber].url} alt="Italian Trulli"
     style={mainPhoto}/>
      <a className="prev"  style={arrowStylePrev} onClick={handlePrev}>&#10094;</a>
      <a className="next"  style={arrowStyleNext} onClick={handleNext}>&#10095;</a>
      <Thumbnail
       productID={productID}
       photoNumber={photoNumber}
       setPhotoNumber={setPhotoNumber}
       styleNumber={styleNumber}
       photoList={photoList}
     />
    </div>

  </>
  )
}


export default MainPhoto;