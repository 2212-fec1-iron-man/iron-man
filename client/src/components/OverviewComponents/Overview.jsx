import React from "react";
import {useEffect, useState} from "react";
import axios from "axios";
import MainPhoto from "./MainPhoto.jsx";
import ProdInfo from "./ProdInfo.jsx";




const Overview = ({productID}) => {

  const [photoNumber, setPhotoNumber] = useState(0);
  const [styleNumber, setStyleNumber] = useState(0);

  const overviewStyle = {
    display: "flex"
  }

  return (
    <div style={overviewStyle}>
     <MainPhoto
        productID={productID}
        photoNumber={photoNumber}
        setPhotoNumber={setPhotoNumber}
        styleNumber={styleNumber}
     />
     <ProdInfo
        productID={productID}
        styleNumber={styleNumber}
        setStyleNumber={setStyleNumber}
      />
    </div>
  )
}

export default Overview;