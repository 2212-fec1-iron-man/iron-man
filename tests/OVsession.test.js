import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/react/dont-cleanup-after-each';
import userEvent from '@testing-library/user-event';
import {render, screen, waitFor, cleanup, fireEvent, waitForElement} from '@testing-library/react';
import Overview from '../client/src/components/OverviewComponents/Overview.jsx';
import MainPhoto from '../client/src/components/OverviewComponents/MainPhoto.jsx';


describe('QAIndex', () => {
  afterEach(cleanup);
  const user = userEvent.setup()

  // it ('snap shots tests for Overview ', () => {

  //   const { container } = render(<Overview produnctID={1} />);

  //   expect(container).toMatchSnapshot();
  // })

  it ('snap shots tests for Checkout ', () => {

    const { container } = render(<MainPhoto styleEntry= {{produnctID: 37311, photoNumber:1, setPhotoNumber: null, styleNumber: 0}}/>);

    expect(container).toMatchSnapshot();
  })



  // it('renders QA head', () => {
  //   render(<QAIndex />);
  //   const head = screen.getByTestId('todo-1');

  //   // console.log("the actual head i got :",head)
  //   expect(head).toHaveTextContent('QUESTIONS & ANSWERS');
  // });




})