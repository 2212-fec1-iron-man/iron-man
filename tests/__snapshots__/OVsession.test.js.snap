// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QAIndex snap shots tests for Checkout  1`] = `
<div>
  <span>
    Data not available
  </span>
</div>
`;
